### PR TITLE
Update pubspec SDK constraints to >=2.1.0-dev.5.0.

### PR DESCRIPTION
**Pull request by stereotype441**
In Dart SDK version 2.1.0-dev.5.0, dart:core was modified so that it
exports the classes Future and Stream.  (Previously these classes were
only available via an import of dart:async).  Some of the useages of
Future and Stream in Flutter fail to import dart:async, therefore in
order to work correctly Flutter requires Dart SDK version
2.1.0-dev.5.0 or later.

The Dart analyzer team is in the process of implementing a hint to
help users remember to update their pubspecs when using Future or
Stream and not importing dart:async
(https://github.com/dart-lang/sdk/issues/35162).  In order to avoid
Flutter failing this hint when it lands, we need to update the
pubspecs now.

### CODE REVIEW

**Review by bwilkerson — APPROVED** [review]
LGTM, and I think this is likely to be the right fix, but given that the other option is to explicitly add an import of `dart:async`, we might want someone on the Flutter team to decide which approach to take.

**Comment by Hixie** [comment]
Honestly we can just remove these SDK constraints -- the SDK that runs for any particular version of those packages is always predetermined and fixed. I think the only reason we have them at all is that without it `pub` used to refuse to do a `pub get` since we were on 2.x ahead of `pub`, but I assume that's been fixed now.

**Comment by Hixie** [comment]
(Please don't commit this before December 10th. We are in a code freeze for risky changes.)

**Comment by stereotype441** [comment]
@Hixie
> Honestly we can just remove these SDK constraints -- the SDK that runs for any particular version of those packages is always predetermined and fixed. I think the only reason we have them at all is that without it `pub` used to refuse to do a `pub get` since we were on 2.x ahead of `pub`, but I assume that's been fixed now.

It turns out that removing the SDK constraints entirely didn't work--pub still treats a missing SDK constraint as implying Dart 1.0, which throws the pub version solver into chaos.  I don't know if this is intentional behavior of pub or if the pub maintainers simply forgot to update the behavior when Dart 2.x was released.

My inclination is to land this CL as is for now.  If pub later changes so that a missing SDK constraint no longer implies Dart 1.0, we can make a follow up CL to remove the SDK constraints.  WDYT?

**Comment by a-siva** [comment]
Is it time to land this PR as we are about to roll Dart forward to the latest bleeding edge and I assume we will hit this SDK constraint issue when that is done.

**Comment by stereotype441** [comment]
> Is it time to land this PR as we are about to roll Dart forward to the latest bleeding edge and I assume we will hit this SDK constraint issue when that is done.

Yes, we will hit this issue the next time Dart is rolled in Flutter, and this PR would address it.  I spoke with Hixie over IM earlier in the week and he would rather not land it in its present form; I think that since Flutter is always locked to a certain version of Dart, he feels that the pubspec's SDK constraints are unnecessary bolierplate, and that the Flutter team will be burdened by having to continually update them in the future.  We discussed several alternatives: remove the constraints entirely (doesn't work because the pub version solver treats the absence of a constraint as requiring Dart SDK < 2.0.0), change the constraints to `any` (wouldn't prevent the hint), leave the pubspecs as is and add imports of `dart:async` to the affected files (would work but wouldn't address the boilerplate issue), land the CL as is (would work but wouldn't address the boilerplate issue), or change the SDK constraints to `any` and add imports of `dart:async` (would work).

Hixie's latest suggestion was to change the SDK constraints to `any` and update `analysis_options.yaml` to suppress the hint, on the grounds that since Flutter is always locked to a certain version of Dart, it receives no benefit from it.  Unfortunately at that point I ran out of time to continue working on the issue; fortunately it should be fairly easy for the person doing the roll to make this change and validate it, so I passed the baton to the Flutter team.

Thinking about it some more, I think there is a disadvantage to this approach: a user might copy and paste Flutter's `analysis_options.yaml` file into a package they intend to publish on pub.  Such a package would not necessarily be locked to a particular version of Dart or Flutter, so it might indeed benefit from the hint, but the user probably wouldn't think to re-enable it when copying over the analysis options.  So I actually think we would serve our users better by landing the CR in its present form.  But since I don't have time to continue working on this, I'll leave it to the Flutter team's discretion how to proceed.

**Comment by a-siva** [comment]
What is the resolution on this? Dart has been rolled into the engine and we are hitting this issue now.

**Comment by goderbauer** [comment]
(cleaning out PR backlog) Is this PR still relevant? It's pretty old and Dart has bee rolled a couple of times. Can we close it?

**Comment by a-siva** [comment]
We have worked around the issue by setting the ignore errors reported for analyzer option 'sdk_version_async_exported_from_core' in the flutter framework sources.
If we want this check to be turend back on I figure some variant of this PR would have to land.

**Comment by goderbauer** [comment]
This change will not affect our users, right? They will not have to bump the sdk version in the pubspec of their app? 

In that case, this change LGTM if it is updated to remove the ignore on the lint.

**Comment by goderbauer** [comment]
Looks like nobody is actually working on this for now. I am going to close the PR for now to reduce the number of stale PRs in our backlog. Feel free to re-open when somebody picks up the work on this again.

**Pull request closed without merging**
